rct229/rules/section5.py

```diff
@@ -97,3 +111,215 @@ def rule_check(self, context, calc_vals, data=None):
+            # Climate zone should have only one in RMR
+            climate_zone = data["climate_zone"]
+            # Becasue list_path is building, the scope of the context.baseline here is building
+            scc_dictionary_b = get_surface_conditioning_category_dict(
+                climate_zone, context.baseline
+            )
+            # Retrieve all surfaces under the building
+            surfaces_b = find_all("$..zones[*].surfaces[*]", context.baseline)
+            # build-in manual check flag.
+            calc_val = {}
+            failing_surface_c_factor_ids = []
+            mix_surface_c_factor_ids = []
+
+            for surface_b in surfaces_b:
+                if get_opaque_surface_type(surface_b) == BELOW_GRADE_WALL:
+                    # construction info
+                    surface_construction_b = surface_b["construction"]
+                    scc_b = scc_dictionary_b[surface_b["id"]]
+                    # QNS a good number to estimate?
+                    target_c_factor = 0.0
+                    # TODO All these need to change to enum later
+                    if (
+                        scc_b == "EXTERIOR RESIDENTIAL"
+                        or scc_b == "EXTERIOR NON-RESIDENTIAL"
+                        or scc_b == "SEMI-EXTERIOR"
+                    ):
+                        target = table_G34_lookup(climate_zone, scc_b, BELOW_GRADE_WALL)
+                        target_c_factor = (
+                            target["c_factor"] if "c_factor" in target else None
+                        )
+
+                        if target_c_factor is None:
+                            raise Exception(
+                                f"Failed finding the c-factor for below grade wall in climate zone {climate_zone}"
+                            )
+
+                    elif scc_b == "EXTERIOR MIXED":
+                        target = table_G34_lookup(
+                            climate_zone, "EXTERIOR RESIDENTIAL", BELOW_GRADE_WALL
+                        )
+                        target_c_factor_res = (
+                            target["c_factor"] if "c_factor" in target else None
+                        )
+
+                        target = table_G34_lookup(
+                            climate_zone, "EXTERIOR NON-RESIDENTIAL", BELOW_GRADE_WALL
+                        )
+                        target_c_factor_nonres = (
+                            target["c_factor"] if "c_factor" in target else None
+                        )
+
+                        if (
+                            target_c_factor_res is None
+                            or target_c_factor_nonres is None
+                        ):
+                            raise Exception(
+                                f"Failed finding the c-factor for below grade wall in climate zone {climate_zone}"
+                            )
+
+                        if (
+                            target_c_factor_res.magnitude
```

Comment:
You shouldn't have to take magnitudes.  Pint can handle `!=`. Since both target values come from our lookup table, they _should_ be the same.